driver/ppm_events_public.h

```diff
@@ -20,6 +20,168 @@ or GPL2.txt for full copies of the license.
+ * kernel versions greater or equal than `2.6`, since this tracepoint has
+ * been introduced in the following kernel release:
+ * https://github.com/torvalds/linux/commit/0a16b6075843325dc402edf80c1662838b929aff
+ */
+#if defined(CONFIG_ARM64)
+	#define CAPTURE_SCHED_PROC_FORK 
+#endif
+
+///////////////////////////////
+// CAPTURE_SCHED_PROC_EXEC 
+///////////////////////////////
+
+/* In some architectures we are not able to catch the `execve exit event` 
+ * from the `sys_exit` tracepoint. This is because there is no 
+ * default behavior among different architectures... you can find more 
+ * info here: 
+ * https://www.spinics.net/lists/linux-trace/msg01001.html
+ * 
+ * Anyway, to not lose this event, we need to instrument a new kernel tracepoint:
+ * 
+ * - `sched_process_exec`: allows us to catch every process that correctly performs
+ *                         an `execve` call.
+ * 
+ * In this way we can send to userspace a `PPME_SYSCALL_EXECVE_X` event
+ * as we do with the `sys_exit` tracepoint.
+ * 
+ * All the architectures that need this patch can use our BPF probe with all 
+ * supported kernel versions (so >= `4.14`), since `BPF_PROG_TYPE_RAW_TRACEPOINT` are
+ * not required in this case.
+ * 
+ * If you run old kernels, you can use the kernel module which requires 
+ * kernel versions greater or equal than `3.4`, since this tracepoint has
+ * been introduced in the following kernel release:
+ * https://github.com/torvalds/linux/commit/4ff16c25e2cc48cbe6956e356c38a25ac063a64d
+ */
+#if defined(CONFIG_ARM64)
+	#define CAPTURE_SCHED_PROC_EXEC 
+#endif
+
+///////////////////////////////
+// CAPTURE_64BIT_ARGS_SINGLE_REGISTER 
+///////////////////////////////
+
+/* This is described in syscall(2). Some syscalls take 64-bit arguments. On
+ * arches that have 64-bit registers, these arguments are shipped in a register.
+ * On 32-bit arches, however, these are split between two consecutive registers,
+ * with some alignment requirements. Some require an odd/even pair while some
+ * others require even/odd. For now, I assume they all do what x86_32 does, and
+ * we can handle the rest when we port those.
+ */
+#ifdef CONFIG_64BIT
+	#define CAPTURE_64BIT_ARGS_SINGLE_REGISTER
+#endif /* CONFIG_64BIT */
+
+///////////////////////////////
+// CAPTURE_CONTEXT_SWITCHES 
+///////////////////////////////
+
+#define CAPTURE_CONTEXT_SWITCHES
+
+///////////////////////////////
+// CAPTURE_SIGNAL_DELIVERIES 
+///////////////////////////////
+
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 32))
+	#define CAPTURE_SIGNAL_DELIVERIES
+#endif
+
+///////////////////////////////
+// CAPTURE_PAGE_FAULTS 
+///////////////////////////////
+
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(3, 12, 0)) && defined(CONFIG_X86)
+	#define CAPTURE_PAGE_FAULTS
+#endif
+
+#else /* Userspace */
+
+/* Please note: the userspace loads the filler table for the bpf probe
+ * so it must define these macro according to what BPF supports
+ */
```

Comment:
This is unfortunate (the bpf engine should take care of this itself and ideally this header would be engine-agnostic), but now that we have a comment about this, maybe we'll have the motivation to fix it ;)